src/coloring/acyclic_coloring.jl

```diff
@@ -152,51 +139,93 @@ end
 Subroutine to merge trees present in the disjoint set which have a
 common edge.
 """
-function merge_trees!(set::DisjointSets{LightGraphs.Edge},
-                    v::Integer,
-                    w::Integer,
-                    x::Integer,
-                    g::LightGraphs.AbstractGraph)
-    edge1 = find_edge(g, v, w)
-    edge2 = find_edge(g, w, x)
-    e1 = find_root(set, edge1)
-    e2 = find_root(set, edge2)
-    if (e1 != e2)
+function merge_trees!(v::Integer,
+                      w::Integer,
+                      x::Integer,
+                      g::LightGraphs.AbstractGraph,
+                      set::DisjointSets{<:Integer})
+    e1 = find(v,w,g,set)
+    e2 = find(w,x,g,set)
+    if e1 != e2
         union!(set, e1, e2)
     end
 end
 
 
+"""
+        make_set!(v::Integer,
+                  w::Integer,
+                  g::LightGraphs.AbstractGraph,
+                  set::DisjointSets{<:Integer})
+
+creates a new singleton set in the disjoint set 'set' consisting
+of the edge connecting v and w in the graph g
+"""
+function make_set!(v::Integer,
```

Comment:
This function name should have a more explicit name. `make_set!` sounds like it builds a new set object, not mutates a disjoint set structure